chia/util/keyring_wrapper.py

```diff
@@ -187,17 +202,25 @@ def set_keys_root_path(keys_root_path: Path):
     @staticmethod
     def get_shared_instance(create_if_necessary=True):
         if not KeyringWrapper.__shared_instance and create_if_necessary:
-            KeyringWrapper.__shared_instance = KeyringWrapper(keys_root_path=KeyringWrapper.__keys_root_path)
+            KeyringWrapper.__shared_instance = KeyringWrapper.create(keys_root_path=KeyringWrapper.__keys_root_path)
 
         return KeyringWrapper.__shared_instance
 
     @staticmethod
     def cleanup_shared_instance():
-        KeyringWrapper.__shared_instance = None
+        if KeyringWrapper.__shared_instance is not None:
+            KeyringWrapper.__shared_instance.cleanup_keyring()
+            KeyringWrapper.__shared_instance = None
+
+    def cleanup_keyring(self):
+        filekeyring: Optional[FileKeyring] = self.keyring if type(self.keyring) == FileKeyring else None
+        if filekeyring is not None:
+            filekeyring.cleanup_keyring_file_watcher()
+        self.keyring = None
 
     @staticmethod
-    def get_legacy_instance() -> Optional["KeyringWrapper"]:
-        return KeyringWrapper(force_legacy=True)
+    def get_legacy_instance():
```

Comment:
Can we add back the return type for this method? Maybe this is another case where TypeVar is appropriate. `get_shared_instance()` should probably specify the return type as well (though I understand if we want to skip that for now to land this PR).

Reply:
Yes, this looks like a merge mistake

Reply:
You can't `TypeVar` without a parameter to relate to the return.  Will have to switch to a `@classmethod` and use `cls`.